.github/workflows/main.yml

```
@@ -5,12 +5,49 @@ on:
+    runs-on: ubuntu-latest
+    outputs:
+      previous_version: ${{ steps.extract_versions.outputs.PREVIOUS_VERSION }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Extract versions
+        id: extract_versions
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2086:info:4:46: Double quote to prevent globbing and word splitting

(shellcheck)